diff --git a/src/plugins/preprocess.js b/src/plugins/preprocess.js
index f6ba8ee39def5b2d3eb99bd93bb455d3bcea17f1..ce5d09823fa4f6400a58ecbbcde232dd64cce8d2 100644
--- a/src/plugins/preprocess.js
+++ b/src/plugins/preprocess.js
@@ -31,18 +31,18 @@ export function preprocess(api) {
 		name: 'vite-plugin-svelte:preprocess',
 		enforce: 'pre',
 		configResolved(c) {
-			//@ts-expect-error defined below but filter not in type
-			plugin.transform.filter = api.filter;
 			options = api.options;
 			if (arraify(options.preprocess).length > 0) {
 				preprocessSvelte = createPreprocessSvelte(options, c);
+				plugin.transform.filter = api.filter;
 			} else {
 				log.debug(
 					`disabling ${plugin.name} because no preprocessor is configured`,
 					undefined,
 					'preprocess'
 				);
-				delete plugin.transform;
+				preprocessSvelte = undefined;
+				plugin.transform.filter = { id: /$./ };
 			}
 		},
 		configureServer(server) {
